integration-tests/opentelemetry-reactive/src/test/java/io/quarkus/it/opentelemetry/reactive/OpenTelemetryReactiveTest.java

```diff
@@ -338,6 +342,45 @@ void multipleUsingCombineDifferentPaths() {
         assertEquals("helloGetUniExecutor", helloGetUniExecutorInternal.get("name"));
     }
 
+    @Test
+    public void potentialTracelessResourceMethods() {
+        when().get("/reactive/potentially-traceless")
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("@Traceless"));
+
+        // should throw because there is no span
+        assertThrows(ConditionTimeoutException.class, () -> {
+            await().atMost(5, SECONDS).until(() -> !getSpans().isEmpty());
+        });
+
+        when().post("/reactive/potentially-traceless")
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("Not-@Traceless"));
+
+        await().atMost(5, SECONDS).until(() -> getSpans().size() == 1);
+    }
+
+    @Test
+    public void tracelessWithClient() {
+        given().when().get("/reactive/traceless-with-client")
+                .then()
+                .statusCode(200)
+                .body(Matchers.is("Hello jack"));
+        await().atMost(500, SECONDS).until(() -> getSpans().size() == 3);
+
+        List<Map<String, Object>> spans = getSpans();
+        //GET traceless-with-client missing
+        assertEquals("helloGet", spans.get(0).get("name"));
+        assertEquals("INTERNAL", spans.get(0).get("kind"));
+        assertEquals("GET /reactive", spans.get(1).get("name"));
+        assertEquals("SERVER", spans.get(1).get("kind"));
+        assertEquals("GET /reactive", spans.get(2).get("name"));
+        assertEquals("CLIENT", spans.get(2).get("kind"));// FIXME this Span refers a sampled out span.
```

Comment:
(We should not assert on get(x)... this is just to illustrate the issue)
Added this test case for a chained call.
What we are doing here is called tail sampling. The OTel Collector is able to do it, but in there, the entire trace is stored and will wait for the parent span to end before deciding if the spans are going to be sent based on a complex set of rules.
Whet er are doing here is much simpler but similar.
The problem here, is that the client span (the 2nd one to be issued) will not have a parent, because it was sampled out.
There are some option we could explore:
* If the trace hits a `@Traceless`... the entire trace (all spans) are discarded
* If a span has `@Traceless`, it and all childs are discarded.